### ISSUE TITLE
Keep textarea in sync with cursor

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#2598]
We only sync the textarea when composition and right click events happen due to an old IE bug where the textarea gripper would always appear. We no longer care around IE support so we should be able to keep the textarea in sync with the cursor which would mean accessibility features like the magnifier app on Windows will work.

VS Code: https://github.com/microsoft/vscode/issues/86107

**Comment by Tyriar** [comment]
Moving cursor code pointers:

https://github.com/xtermjs/xterm.js/blob/8b5c48fe878f830914248e4c1ed78e621deb16c4/src/browser/input/CompositionHelper.ts#L209-L228

https://github.com/xtermjs/xterm.js/blob/8b5c48fe878f830914248e4c1ed78e621deb16c4/src/browser/Clipboard.ts#L72-L91

When to update the cursor's position could be done in a few places, listening to `InputHandler.onCursorMove` would probably be a good start:

https://github.com/xtermjs/xterm.js/blob/8b5c48fe878f830914248e4c1ed78e621deb16c4/src/Terminal.ts#L280

**Comment by BuraChuhadar** [comment]
I would like to work on this please if possible.

**Comment by BuraChuhadar** [comment]
Here is my first attempt: https://github.com/xtermjs/xterm.js/pull/2632. I essentially brought the compositionhelper code for updating the cursor position to keydown event in the terminal. I tested on my side even though it is very hard to test it, I believe it is working as expected.

**Comment by jeanp413** [comment]
I'll give this a shot
